### PR TITLE
`azurerm_static_site` - `identity` supports `SystemAssigned, UserAssigned`

### LINKED ISSUE #15794: Support for both UserAssigned and SystemAssigned for Static Sites
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Currently, azurerm_static_site only support either UserAssigned or SystemAssigned, but not both as with azurerm_function_app and azurerm_app_service. Please provide a means to declare both.

### New or Affected Resource(s)

* azurerm_static_site

### Potential Terraform Configuration

```hcl
identity {
  type = "SystemAssigned, UserAssigned"
  identity_ids = ...
}
```

### References

* #14878

### PR DESCRIPTION
**Pull request by magodo**
Fix #15794

## Test

```shell
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/web  -run='TestAccAzureStaticSite_with|TestAccAzureStaticSite_identity'
=== RUN   TestAccAzureStaticSite_withSystemAssignedIdentity
=== PAUSE TestAccAzureStaticSite_withSystemAssignedIdentity
=== RUN   TestAccAzureStaticSite_identity
=== PAUSE TestAccAzureStaticSite_identity
=== RUN   TestAccAzureStaticSite_withSystemAssignedUserAssignedIdentity
=== PAUSE TestAccAzureStaticSite_withSystemAssignedUserAssignedIdentity
=== RUN   TestAccAzureStaticSite_withUserAssignedIdentity
=== PAUSE TestAccAzureStaticSite_withUserAssignedIdentity
=== CONT  TestAccAzureStaticSite_withSystemAssignedIdentity
=== CONT  TestAccAzureStaticSite_withSystemAssignedUserAssignedIdentity
=== CONT  TestAccAzureStaticSite_identity
=== CONT  TestAccAzureStaticSite_withUserAssignedIdentity
--- PASS: TestAccAzureStaticSite_withSystemAssignedUserAssignedIdentity (131.75s)
--- PASS: TestAccAzureStaticSite_withSystemAssignedIdentity (182.29s)
--- PASS: TestAccAzureStaticSite_identity (227.74s)
--- PASS: TestAccAzureStaticSite_withUserAssignedIdentity (807.10s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/web   807.112s
```

### CODE REVIEW

File: internal/services/web/static_site_resource_test.go [hunk 1]
```diff
@@ -77,13 +77,14 @@ func TestAccAzureStaticSite_identity(t *testing.T) {
...
 		},
 		data.ImportStep(),
-		{
-			Config: r.basic(data),
-			Check: acceptance.ComposeTestCheckFunc(
-				check.That(data.ResourceName).ExistsInAzure(r),
-			),
-		},
-		data.ImportStep(),
+		// There is API issue to downgrade the sku to Free with identity configured: https://github.com/Azure/azure-rest-api-specs/issues/18253
```
Comment (tombuildsstuff) [inline internal/services/web/static_site_resource_test.go:80]: can we make this a TODO:

```suggestion
		// TODO: re-enable this once the API issue is resolved: https://github.com/Azure/azure-rest-api-specs/issues/18253
		// API currently doesn't allow downgrading to a Free SKU with an Identity attached
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
A couple of minor comments but if we can fix those up then this should otherwise be good to go 👍

File: internal/services/web/static_site_resource_test.go [hunk 2]
```diff
@@ -77,13 +77,14 @@ func TestAccAzureStaticSite_identity(t *testing.T) {
 			),
 		},
 		data.ImportStep(),
```
[the remark this answers is not in the document]
Comment (magodo) [reply internal/services/web/static_site_resource_test.go:79]: @tombuildsstuff Isn't the current step is the SystemAssinged and UserAssigned use case? While I noticed that the 3rd and 5th steps are actually the same, I'm not sure whether it is intentional, so just keep it as is.

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply internal/services/web/static_site_resource_test.go:79]: Yep, good spot - yes the 5th should be UserAssigned I guess?

**Comment by magodo** [comment]
@tombuildsstuff I've updated the test and passed it, please take another look, thank you!

## Test

```shell
💤  TF_ACC=1 go test -v -timeout=20h ./internal/services/web  -run='TestAccAzureStaticSite_identity$'
=== RUN   TestAccAzureStaticSite_identity
=== PAUSE TestAccAzureStaticSite_identity
=== CONT  TestAccAzureStaticSite_identity
--- PASS: TestAccAzureStaticSite_identity (873.12s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/web   873.134s
```

**Pull request merged**